.circleci/config.yml

```diff
@@ -1,158 +1,165 @@
     steps:
       - checkout
+      - restore_cache:
+          keys:
+            - influxdb-cache-v1-{{ checksum "go.mod" }}
+            - influxdb-cache-v1
       - run:
-          name: Check OSS release candidacy
+          name: Execute tests
           command: |
-            echo "TODO: Run checks to determine if this build is a release candidate"
-            ls ./
+            set -x
+            mkdir -p junit
+            go test -v ./... 2>&1 | tee tests.log
+            go-junit-report < tests.log > junit/influxdb.junit.xml
           no_output_timeout: 1500s
-  test:
-    machine:
-      enabled: true
-      docker_layer_caching: true
-    environment:
-      - PARALLELISM: 4 # Amount of parallelism within each test (not the number of tests)
-    parallelism: 3 # How many CircleCI test containers
+      - store_test_results:
+          path: junit/
+  unit_test_tsi:
+    docker:
+      - image: quay.io/influxdb/cross-builder:<< pipeline.parameters.cross-container-tag >>
+    resource_class: large
     steps:
-      - attach_workspace:
-          at: /tmp/workspace
-      - source-vars
       - checkout
+      - restore_cache:
+          keys:
+            - influxdb-cache-v1-{{ checksum "go.mod" }}
+            - influxdb-cache-v1
       - run:
-          name: Ensure CircleCI parallelism matches "./test.sh count"
-          command: "[ `./test.sh count` -eq $CIRCLE_NODE_TOTAL ]"
+          name: Execute tests
+          command: |
+            set -x
+            mkdir -p junit-tsi
+            export INFLUXDB_DATA_INDEX_VERSION="tsi1"
+            go test -v ./... 2>&1 | tee tests.log
+            go-junit-report < tests.log > junit-tsi/influxdb.junit.xml
+          no_output_timeout: 1500s
+      - store_test_results:
+          path: junit-tsi/
+  unit_test_race:
+    docker:
+      - image: quay.io/influxdb/cross-builder:<< pipeline.parameters.cross-container-tag >>
+    resource_class: large
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - influxdb-cache-v1-{{ checksum "go.mod" }}
+            - influxdb-cache-v1
       - run:
-          name: Execute test
-          command: ./test.sh $CIRCLE_NODE_INDEX
+          name: Execute tests
+          command: |
+            set -x
+            mkdir -p junit-race/
+            export GORACE="halt_on_error=1"
+            go test -race -v ./... 2>&1 | tee tests.log
+            go-junit-report < tests.log > junit-race/influxdb.junit.xml
           no_output_timeout: 1500s
+      - store_test_results:
+          path: junit-race/
   fluxtest:
-    machine:
-      enabled: true
-      docker_layer_caching: true
+    docker:
+      - image: quay.io/influxdb/cross-builder:<< pipeline.parameters.cross-container-tag >>
     steps:
       - checkout
       - run:
           name: Execute test
-          command: ./test.sh flux
+          command: ./test-flux.sh || exit 1
```

Comment:
Is the `|| exit 1` needed here? I would have thought the job would fail on a non-zero exit code from the `test-flux.sh` script without that.

Reply:
This is in case someone adds some more lines to the script - as written it doesn't matter.